### ISSUE TITLE
Zak's Design Tweaks & Rewrites Thread

### ISSUE THREAD

**Issue by caycefischer** [HabitRPG/habitica#585]
Just a log for me to record before/after screencaps of my random interface tweaks. Anyone is welcome to suggest further changes or critique.

This is probably only of interest to UI nerds and members of the dev team.

**Comment by caycefischer** [comment]
#### Commit ba1a8dabc854b286575ef11bb7acbf9ff07badb1
- Moving styles away from bootstrap
- start implementing Lato as app-wide font (fuck helvetica)
- normalize font sizes and padding/margin/radii using relative `em` units
- begin refactoring of tasks html (remove pre element, add module classes)

---

Before:
![Screen Shot 2013-03-04 at 8 33 52 PM](https://f.cloud.github.com/assets/546495/219537/ca6b734e-8535-11e2-86fe-edf07a9d3fef.png)

After:
![Screen Shot 2013-03-04 at 8 33 42 PM](https://f.cloud.github.com/assets/546495/219539/cefa7a68-8535-11e2-8445-9da4238fec51.png)

---

##### Notes to self:
- Reduce letter-spacing on task text

**Comment by switz** [comment]
I like helvetica :\

**Comment by caycefischer** [comment]
@switz I have nothing against helvetica intrinsically, but it's become so genericized in design and especially thanks to bootstrap, it's like the times new roman of the web now. It's not really a serious option anymore, in this kind of  minimal application—regrettable, but true. Lato might not be the right option, I'm open to suggestions.

**Comment by caycefischer** [comment]
Apple actually uses Myriad Pro for its own branding, but you're absolutely
right—in the era of OSX Helvetica came bundled with the system and became
one of the default sans-serif fonts on desktops (and the web). But the
popularity of Helvetica as the standard "good design" font has been strong
for at least a few decades more than the internet age. It was originally a
heading/signage/poster font, and more recently, as you say, has been
popularized as a body font due to the digital ubiquity.

On Mon, Mar 4, 2013 at 9:44 PM, Rudd Fawcett notifications@github.comwrote:

> @zakkain https://github.com/zakkain @switz https://github.com/switzThis may just be me, but I feel like Apple popularized Helvetica, and more
> people started to use it... Might just be me, though...
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/lefnire/habitrpg/issues/585#issuecomment-14420454
> .

## 

_– Zachary Kain_

Designer, Creative Technologist
416-712-8895
zakkain@gmail.com
→ view my resume/profile http://zerply.com/zakkain/public

**Comment by wildcate** [comment]
Looks nice! Personally, I rather like Trebuchet (even if it was designed for Microsoft), but I'm not sure how well it would serve for this purpose.

**Comment by lefnire** [comment]
Some thoughts:

Let's not go too crazy gutting Bootstrap until we're rich. Here's why. The files come in heavy, but they're the least of our optimization concerns in Habit - trust me. We use tons of their styles, so it's gonna be a major overhaul and extra project to maintain replacing them. It's not as simple as replacing the styles, as we saw with the progress-bars scare. Also, we use almost _all_ of their javascript components, which depend on the styles anyway - so we'd be sending the css to the client regardless (we're definitely not rewriting their JS components). Add & alter as necessary, but let's not gut - not till we have $$.

Re fonts: I trust Zach's opinion here because he does this professionally, but we just heard 4 font opinions, in which case I'm inclined to go with the opinion of Twitter (Bootstrap - aka, Helvetica). But let's give Zach's commits a shot and see what people think.

Side note - let's make all non-hotfix commits as pull requests, it makes merge-sets much easier to track. I'm gonna start doing this myself as well.

Hope this doesn't sound like a combo-breaker. You truly are the man, Zach!

**Comment by caycefischer** [comment]
@lefnire Really good point re: bootstrap—I won't gut anything, I'll just layer new CSS on top. It gets compressed anyway so like you said, it isn't an optimization concern.

You're going to get pushback whenever you make changes, and people are used to the Bootstrap look. I'm certainly open to font changes but I'm campaigning pretty hard for ditching helvetica. Let's see if there's mass complaining or not.

OK, good point as well. So, how should I do this? How do you pull request commits from the same repo? Or should I maintain my fork, and just PR stuff in from zakkain/habitrpg and merge it?

**Comment by lefnire** [comment]
Fair, I'll keep take complaints with a grain of salt (you're absolutely right, the push back has already begun now we have established users). I'll support you on the font change then my friend.

Don't worry abou this merge set, keep in mind for next batch. Branches. You can do it either zakkain/habitrpg, you can even push to lefnire/habitrpg#new-branch and PR it against develop (that's how I'm doing it). Whatever's easiest.

**Comment by caycefischer** [comment]
Oh OK! I'll make a ui-tweaks branch or something and PR across branches.

On Tue, Mar 5, 2013 at 8:34 AM, Tyler Renelle notifications@github.comwrote:

> Fair, I'll keep take complaints with a grain of salt (you're absolutely
> right, the push back has already begun now we have established users). I'll
> support you on the font change then my friend.
> 
> Don't worry abou this merge set, keep in mind for next batch. Branches.
> You can do it either zakkain/habitrpg, you can even push to
> lefnire/habitrpg#new-branch and PR it against develop (that's how I'm doing
> it). Whatever's easiest.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/lefnire/habitrpg/issues/585#issuecomment-14440010
> .

## 

_– Zachary Kain_

Designer, Creative Technologist
416-712-8895
zakkain@gmail.com
→ view my resume/profile http://zerply.com/zakkain/public

**Comment by switz** [comment]
Just to reiterate Tyler for cleanliness' sake, fork the repo (button at top right) and make changes on a new branch in your repo. Then send a pull request from your repo to the develop branch on tyler's repo. This makes life much easier when working with a team.

**Comment by caycefischer** [comment]
@lefnire Thanks Tyler—Lato might not be right in the end, but the helvetica non-choice is no better :P

@switz aha OK. Yes, that does make it clearer. Wilco.

@lefnire And to be clear just so I don't muck this process up, d'you want to be the guy managing PRs or should I merge them myself?

**Comment by lefnire** [comment]
Me please :) Just so I can do quick-scan checks pre-merge

**Comment by caycefischer** [comment]
![byyourcommand](https://f.cloud.github.com/assets/546495/221455/d71ede7a-859c-11e2-850e-e734d4ea03d0.gif)

**Comment by lefnire** [comment]
Woo! I haven't gotten a zakkain special in a while, forgot how awesome those are!

**Comment by caycefischer** [comment]
I'm working on better checkboxes and buttons, for a few reasons:
- moves us from the default browser UI and makes the app more unique, something that's identifiable as "our style"
- that will be useful in the mobile app too, to get a nice "native" feel
- improve accessibility and visual aesthetics

So, here they are so far:

![Screen Shot 2013-03-05 at 12 53 32 PM](https://f.cloud.github.com/assets/546495/223907/8e9ce84e-85c5-11e2-840c-82dee98fbf8e.png)

*\* I want feedback on the faded-out checkmark on uncompleted tasks. When that area is blank it looks really weird, and I think this lets users know you can interact with it. Are they not subtle enough? Too subtle? Maybe replace them with an "X"?**

---

Secondly, I've been extending this style to the other buttons, currently the Habit column. Things are a little  different here, as the PLUS and MINUS buttons currently have a green/red colour to help show what they do, but that can clash with the color of the actual task. See two options below:

![Screen Shot 2013-03-05 at 1 43 41 PM](https://f.cloud.github.com/assets/546495/223921/19c48a1c-85c6-11e2-9bc8-8253dad1eeff.png)![Screen Shot 2013-03-05 at 1 44 57 PM](https://f.cloud.github.com/assets/546495/223922/1e58bf26-85c6-11e2-99a8-3fc114736118.png)

**I want feedback on which of the two techniques you think works better, or if you have suggestions for a better way**. Personally, I like the buttons when they're the same color as the task, it looks a lot more harmonious, and I don't think you need the green/red to help you figure out which is better.

**Comment by Pandoro** [comment]
First of all, I like this style! But it does change the general UI a lot I
guess? We should see how users react to this as well. Give it a while on
the test server maybe ?
1. Too subtle in my opinion. I don't really see the difference between
   checked and not. Maybe try a circle for still open task? A check more for
   done. Furthermore I think the disabled grey tasks aren't so clear. I would
   somehow distinguish them stronger from tasks that need to be completed.
2. I like the second one better as well :) I would round of the edges to
   the left though, as with the others. (Probably this was intended right?)

**Comment by wc8** [comment]
I like the big shaded checkboxes. 
1. I thought they were all checked and wondered why they weren't all grey. Perhaps a faded ☐ symbol in place of the faded checkmark?
2. The right hand/second one looks cleaner, clearer. 
The mix of green and red on the left/first one seems strange.

**Comment by caycefischer** [comment]
@Pandoro _good_ points!

Definitely won't push this live right away, there's a lot of UI left to go over so yes—a testing server would be great to get some opinions. 

Glad you agree on point **2**. I'll try with some different symbols, an empty circle/square might be really nice. :D

**Comment by lefnire** [comment]
Stunning, I love this! I don't have too much advice, but I really like where this is going

**Comment by Pandoro** [comment]
Btw, regarding the rounded edges, we might also go with the trend of sharp
edges :p Like android and windows x)

**Comment by wildcate** [comment]
I like the empty/checked checkboxes and the same colour of all the task boxes of the current version a lot - better than the two-tone version of the mockup. Though there is more choice for strong colours when one can fade it for the text background... and that is nice, too. (Proper angry red for those languishing tasks!)
How about adding an outline to each task, in the stronger colour? Like so:

![outlined](https://f.cloud.github.com/assets/3587347/224089/66cb747e-85cb-11e2-82a2-9232eef24e01.png)

And yes, I also thought that the light checkmark was a sign for "done" at first. 

For the +/- habits, I like the right-side column better - the darker shading of the button part sort of overrides the shading of the habit itself, so it looks like they are just red or just green. Especially if the todo lists and dailies will have the two-tone shading with the darker version of the task colour in front.

Edit: Now that I've looked at these nice strong solid colours while meddling with the mockup, on returning to the UI as is it seemed awfully pale. So :+1: for the new, more solid stuff!

**Comment by caycefischer** [comment]
@wildcate that looks pretty chunky, I dunno. To me that UI feels heavier without really needing it.

@Pandoro I'm a _big_ fan of sharp corners, but I think in this case it's a minor aesthetic thing. Your thoughts on these variations?

Rounded:
![Screen Shot 2013-03-05 at 3 09 04 PM](https://f.cloud.github.com/assets/546495/224277/afbecdf2-85d0-11e2-80dd-1b7848dc74f4.png)

Square:
![Screen Shot 2013-03-05 at 3 09 48 PM](https://f.cloud.github.com/assets/546495/224278/b007eeb0-85d0-11e2-9507-ee87180ee37c.png)

Square with subtle border:
![Screen Shot 2013-03-05 at 3 12 16 PM](https://f.cloud.github.com/assets/546495/224292/09d56dbe-85d1-11e2-9ba3-487a58e8516c.png)

**Comment by Shaners** [comment]
Sharp corners are really in this season!

**Comment by lefnire** [comment]
Actually getting very excited for this!

**Comment by caycefischer** [comment]
Hokay folks, switching tacks for a second to talk about responsive design! Currently the responsive breakpoints are based off of bootstrap's defaults, which for our four-column layout is pretty bad and makes screen sizes just above the breakpoint impossible to use.

Here's my first crack and new responsive styles (excuse the low quality gif):

![rspnsv](https://f.cloud.github.com/assets/546495/225235/9aa2c434-85ec-11e2-8ebe-3a842b537376.gif)

I made breakpoints based on the _content_, so that when things get too cramped, we switch to the best layout. **What do people think of the two-column layout at smaller screen sizes? Do you like the scrolling inside a section when it gets too long (so there aren't big empty spaces), or do you find that confusing?**

![rspnsv](https://f.cloud.github.com/assets/546495/225275/6582a052-85ed-11e2-9588-40641b0864e2.gif)

**Comment by lefnire** [comment]
_drool_

**Comment by caycefischer** [comment]
### [Bleeding-Edge Preview Server](http://habitrpg.herokuapp.com/)

OK, so Tyler and I have decided to throw up a live preview server for these new UI changes. What you see here is 100% bleeding-edge, it _will_ be broken, there _will_ be bugs, designs _will_ be half-baked. Code gets pushed here more or less as I write it.

Feel free to check it out, post bug reports and suggestions in this thread.

**Comment by lefnire** [comment]
looks like bootstrap components nested within effected elements inherited some styling: [before](http://gyazo.com/fb16d6bcb1be54201ac848a65c76683b.png?1362536081) | [after](http://gyazo.com/8a324231733b0cfdf0b7f421cf5b1c57.png?1362536151). Not a biggie, and more something for my own plate - move all modals / popovers outside of partials & pass in data with JS, rather than instantiating components for each task

**Comment by wildcate** [comment]
Hey wow! Sleeping makes one miss things!
- at a few window sizes, the gold/silver icons move over the "Rewards" text and are thus intelligible; the "add new" boxes are wider than the columns they sit in - both these are also issues in the current version
- I like the boxes with border better than those without (makes it seem more like one item and not like a two-part thingie), and the rounded edges better than the sharp corners
- I totally love being able to scroll inside a list when it's too long (my to-do is waaaay longer than the others)

Oh, and regarding the chunky borders I drew in... my tasks are solid, chunky tasks or want to seem like that ;)  

Edit: Forgot to mention that the font did not work for me - too irregular/thick-and-thin looking on my screen.

**Comment by wc8** [comment]
I like the look overall. I personally think the sharp corners make a nice contrast with the circle, but that's a minor detail.

Perhaps already planned, but could the box design be expanded to the rewards section for consistency? 
Maybe the gold cost can be centered in a box matching the checkmarks/plus buttons.

Usability wise, the checkmark on mouseover is a good touch for helping the user know what that box will do. :+1: 
The plus and minus buttons are not as clear, but that's something that can be taken care of by tooltips perhaps, and ultimately comes from knowing how HabitRPG works. Idea: moused-over habit button to change color based on its effect for a visual cue: the **-** button turn red and the **+** button turn green.

Awkwardnesses:
- Reward Price box wide for a number input box; too wide to fit when in 4 column view (Gold label not visible). 
- I can scroll far below the bottom of the main page when the 2x2 view is in effect.
- I think the odd alignment/wrap of long task/reward names is probably covered by @ruddfawcett's comment.
- I've died twice when the page refreshed and it's unclear why, HP seemed good.
- already present issues: habit note popups aren't fully visible in the 4 column width (although it shows up nicely in 2x2 arrangement), and the known algorithm issues

**Comment by Pandoro** [comment]
See my comments in the image. Overall it looks nice, but after seeing it, I am longing back a little to the good old habitrpg feeling. But well I guess that is something you get during every change ^^; 

![bugs](https://f.cloud.github.com/assets/1872333/227210/a9e9c064-8655-11e2-8d6b-d42ec36c6fe9.png)

**Comment by wildcate** [comment]
I have looked a bit more at the old vs. the new version - the new design takes up a lot of space with the buttons/checkpoint area; having just the little box is using much less space. I also still get the impression that the button/checking area is the main item with its darker colour and the lighter-shaded actual task sort of fades into the background. Even if you did not like my chunky borders  ; )  I think they made it very clearly one item visually. Now my eye gets drawn away from the task text and toward the button area.

There's also some light grey box-like thingies around the daily and to-do list items that are missing for the others. They appear when an item is checked.

**Comment by wc8** [comment]
My circles do not look like @Pandoro's. Mine are a nice ratio to the box size. Perhaps a browser/OS difference?

**Comment by Pandoro** [comment]
I'm on Ubuntu 12.04 FF 16.0.2

I hate to say it @zakkain and I really like you as a person, but giving it a little more thought and especially comparison, I like the traditional view better :x 

I think by itself, your design looks really awesome, it looks clean and stylish, it would fit any current system. The problem I have with it though, is that Habitrpg, for me, has a kind of nostalgic feeling. I am not sure what makes the real difference, but if I look at the current design, I get a feeling of a  90s RPG. Rounded corners + pixel art. Simple UI structures, because they did not have anything fancy back then. Now if I look at your design, it just SCREAMS modern in my face. And while I seriously respect that someone can just pull that of in a day or two, it does not feel right in habitrpg for me. 
With the original version, you see the header and you look down and the general style continues and you get the idea that it fits. With the new version, the header seems nice and all, then you look down and you have the idea you are looking at some other website, it simply doesn't fit the pixelish style so well. 

Now if it is just me, please go ahead, push it onto the live server once the bugs are fixed, but honestly I would be cautious here. Many people play this because it brings them back part of their childhood feelings I think. If you start taking that away from them... not so good I think :x  I would fix it up a little and then do a poll, see what people like. Or... the other option, which I would right now really feel for, redesign the header as well. Tear out everything into "theme" files and make it selectable in the menu. 

Just to say it again :x Please don't hate me for this @zakkain, I just want to get my honest opinion out here.

**Comment by caycefischer** [comment]
_(circles are kinda different in every browser, it's a weird line-height thing. It'll get changed to a more reliable method)_

@Pandoro I certainly take your point, there is something fast-talking and modern about the new interface that perhaps lacks some of the original... innocence. But I think the larger point (which I feel too) is that we have some nostalgia for the the original interface, and the whole app is kind of built on nostalgia in a way. Any progress we make is going to have resistance, from ourselves and others, that's just how it goes.

Now, the old UI had a lot of problems and we do need to move forward—but it is still very much a work in progress, and I'll certainly try my best to bring back some of that '90s vibe for ya :)

![freshprince](https://f.cloud.github.com/assets/546495/227449/5961b23c-865f-11e2-8c30-e0583e517a2a.gif)

PS: @Pandoro holy crap that's a lot of XP! :sob:

**Comment by wc8** [comment]
Strangely (since I like user customizations), I'm not a fan of the theming idea at this stage. It seems like HabitRPG should stay uniform for awhile to develop its own character. 

Maybe to emphasize Habit+RPG, the buttons/checkboxes should be even more game-like than the original ones? Or perhaps just the checkmarks and circles within the new theme could get a facelift? Game-style buttons? Bars of gold with coin amounts on the reward button?

Just my 2¢ brainstorming. Looking forward to see what comes next.

**Comment by caycefischer** [comment]
@wc8 I agree.

There's actually a parallel UI in the pipe that's exactly what you're talking about—an optional "skin" you'll be able to toggle to take the app from the basic clean interface to a retro-style game interface. Some talk about it sprinkled in the [Interface & Aesthetics](https://github.com/lefnire/habitrpg/issues/361) thread.

Our goal for the basic design is to keep it as clean and simple as possible, the vast majority of people who've chimed in seem to want a beautiful _utilitarian_ interface to GTD, at least to start with.

**Comment by wc8** [comment]
Just in case, my button suggestions were for just a small hint of game quality; and that was just brainstorming. I agree that utilitarian is really important for HabitRPG. Fun shouldn't come at the cost of clarity or performance. 

[P.S. Catching up on that Interface thread. Thanks for pointing it out. I can see myself switching back & forth between a Clean-Modern basic skin and a future Enhanced Skin in different circumstances.]

One really good benefit of the new design: 
I often zoom out to see more tasks at once (I have long lists). The large areas for clicking work better on zoom out than the old version. With the old version it's hard to get the right spot for that tiny checkbox to be clicked. With the new version, I have no trouble. Although when very zoomed-out there are alignment/visual issues with the circles and checkmarks, I can still click anywhere in the shaded version and register the completion. I really appreciate that.

**Comment by PhilipHow** [comment]
This has also just become free

http://designmodo.com/flat-free/

**Comment by caycefischer** [comment]
aaactually, they're currently embroiled in a DCMA dispute! O_o Their
original repo got taken down, looks like the just moved it.

On Fri, Mar 8, 2013 at 5:36 AM, Philip How notifications@github.com wrote:

> This has also just become free
> 
> http://designmodo.com/flat-free/
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/lefnire/habitrpg/issues/585#issuecomment-14613250
> .

## 

_– Zachary Kain_

Designer, Creative Technologist
416-712-8895
zakkain@gmail.com
→ view my resume/profile http://zerply.com/zakkain/public

**Comment by PhilipHow** [comment]
Indeed, all the more reason to use them.

**Comment by lefnire** [comment]
@zakkain make sure to do a merge of develop, lotta changes in there. Also we're going to have a beta server in the near future with @alanthing's migration to aws. we'll debut your redesign on the beta server, which will track `develop` (while habitrpg.com tracks `master`) and use the production database. Twill fulfilling the KS beta promise, shard the server load, and allow the "dangerous livers" to kick the redesign tires in production while harbinging it's arrival to prod.

**Comment by ghost** [comment]
So we are going to use circles instead of checkmarks now?

**Comment by caycefischer** [comment]
@lefnire Wilco, I keep forgetting to pull that in. Planning on finishing the Rewards column and doing the PR tomorrow.

That beta server is a rad idea!

@Neohuman circles just indicate a task requires completion. It might change to an outlined square, still figuring this stuff out.

**Comment by caycefischer** [comment]
### Update before the hangout tomorrow.

##### 1. Rewards

![Screen Shot 2013-03-09 at 1 37 38 AM](https://f.cloud.github.com/assets/546495/239371/64a89a74-8886-11e2-95f3-e5797f7d307c.png)![rewards min](https://f.cloud.github.com/assets/546495/239372/6c8d3538-8886-11e2-82e2-6662e6f428d4.gif)

**I've brought the rewards and pets over to the new design. Several points:**
- this layout for Pets is only temporary, pending the major re-design and implementation of the Pet/Mount system currently underway

![Screen Shot 2013-03-09 at 1 37 56 AM](https://f.cloud.github.com/assets/546495/239369/57401c2c-8886-11e2-9f8d-50932edbf8ef.png)
- the same goes for the Store, which at some point will become actual Store and Inventory screens
- we need to solve the "shit isn't centered in our sprites" problem, so this doesn't happen:
  ![Screen Shot 2013-03-09 at 1 37 46 AM](https://f.cloud.github.com/assets/546495/239370/60e5a616-8886-11e2-92bc-10eb90563da1.png)
  because that's ugly as sin. Maybe we solve it by having a sprite for the avatar where items are in their proper position, and a separate one for the store? Maybe we just write better store CSS positioning? I dunno.

##### 2. What do y'all prefer: circle or shaded square for Uncompleted Task?

My money is on squares. Not sure if I like it when the square is still visible when checked off, or not.
![Screen Shot 2013-03-09 at 1 37 26 AM](https://f.cloud.github.com/assets/546495/239374/d0b1e37e-8886-11e2-8dd4-648e1f2e7c0b.png)![Screen Shot 2013-03-09 at 1 37 20 AM](https://f.cloud.github.com/assets/546495/239375/d61e9abe-8886-11e2-9e3e-65fbc40c5eb0.png)
![Screen Shot 2013-03-09 at 2 08 33 AM](https://f.cloud.github.com/assets/546495/239385/3dd247e0-8888-11e2-9d7c-787afb07247b.png)

##### 3. Interactions

I've started digging into the jqueryUI stuff and it's a pain in the ass. Here's some sexy better drag interactions (sorry for the bad gif):
![newdrag min](https://f.cloud.github.com/assets/546495/239376/eb656df8-8886-11e2-80b3-3546d70e363a.gif)
I personally didn't like the opacity, this makes more sense. Upcoming on my agenda is figuring out how to get touch events working and setting up sortable task lists for touch screens, not just desktop. That probably means not using the jQueryUI sortable module. We'll see. If it isn't easy or convenient I won't rock the boat.

**Comment by lefnire** [comment]
Fantastic! Pushing to Heroku now, should be up in a few minutes. I vote
boxes too.

**Comment by caycefischer** [comment]
##### Note to self:

Just decided, on a whim, you know, to test on Firefox (durp) and found huge bugs with the checkboxes and all the `outline`s I'm using:
![ffbug](https://f.cloud.github.com/assets/546495/240055/38e6188c-88ed-11e2-8796-2a0e24dba3de.jpg)

**Comment by lefnire** [comment]
also "cost" input on rewards: [screenshot](http://gyazo.com/b5c912ceebd96bba6470a3bd0080ba93.png?1362857161)

**Comment by horusofoz** [comment]
Similar to above for Custom Day Start field.
[Screenshot](https://f.cloud.github.com/assets/3202339/268020/60bc9766-8ef9-11e2-9b7e-81c62949366d.png)

**Comment by horusofoz** [comment]
Also I noticed all progress/history is missing. Sorry if I missed the memo but is this by design?

**Comment by caycefischer** [comment]
Yes, it is – currently stats/ charts are hella broken, so I removed the
button

On Sunday, March 17, 2013, horusofoz wrote:

> Also I noticed all progress/history is missing. Sorry if I missed the memo
> but is this by design?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/lefnire/habitrpg/issues/585#issuecomment-15021480
> .

## 

_– Zachary Kain_

Designer, Developer
Writer, Educator
416-712-8895
zakkain@gmail.com
➝ http://draftingcode.com

**Comment by horusofoz** [comment]
When adding new tasks, the text is visible behind the add button. Applies to Habits/Dailys/Todos/Rewards.
[Screenshot](https://f.cloud.github.com/assets/3202339/269063/b207f2d2-8f76-11e2-9793-48a4690d26d1.png)

**Comment by caycefischer** [comment]
task button opacity fixed in ba16338bf9c83921e9d975229a70a1a8275edea1

input height fixed in eceae48b2bcf45f79da62696dd539867224dcfb3

**Comment by caycefischer** [comment]
Thoughts on a better icon for uncompleted Dailies and Todos? A good point was made during the podcast that the sketched-in box kinda looks like it's completed—filled in. 

Here's a blank one:
![Screen Shot 2013-03-18 at 5 28 36 PM](https://f.cloud.github.com/assets/546495/272920/f8a22518-9012-11e2-8143-86875b0a0721.png)

Here's a circle again:
![Screen Shot 2013-03-18 at 5 30 02 PM](https://f.cloud.github.com/assets/546495/272923/05e7d2ae-9013-11e2-9cc4-8713eb568f12.png)

And some others:
![Screen Shot 2013-03-18 at 5 33 02 PM](https://f.cloud.github.com/assets/546495/272941/782c4d86-9013-11e2-9b2f-2bd14cdf50c2.png)![Screen Shot 2013-03-18 at 5 34 51 PM](https://f.cloud.github.com/assets/546495/272953/b42ec980-9013-11e2-8e42-cdaf77ea6348.png)![Screen Shot 2013-03-18 at 5 36 12 PM](https://f.cloud.github.com/assets/546495/272962/e18d8a06-9013-11e2-8523-163827c18903.png)

---

Personally, I think the top one, the basic square outline, is the way to go. Though there's something cool about the  faux-depth of the bottom left one.

**Comment by switz** [comment]
I vote for the 'totally blank's, I think they look sleek.

**Comment by horusofoz** [comment]
+1 for  'totally blank's, I think they look sleek.'.

Cleanest and easiest to understand IMO.

**Comment by Shaners** [comment]
I like the blank box one (top) seems straight forward and tells me immediately I could click it. 

Maybe the hashed boxes could be used for tasks that aren't able to be completed yet? I.e. if / when stepped tasks are introduced the tasks that require another to be completed first use a hashed box.

**Comment by caycefischer** [comment]
Wow, a lot of support for the blank one! I'm surprised—personally it feels like there's something missing.

@Shaners I like that idea, as we build out more functionality like timed tasks whatnot, we can expand our icon vocabulary.

Well, if a few more people pipe up in favor of the blank box, I'll push and merge to develop in time for the UI going live on the master branch.

**Comment by horusofoz** [comment]
Just to clarify this is the blank one I was referring to:
![https://f.cloud.github.com/assets/546495/272953/b42ec980-9013-11e2-8e42-cdaf77ea6348.png](https://f.cloud.github.com/assets/546495/272953/b42ec980-9013-11e2-8e42-cdaf77ea6348.png)

**Comment by wc8** [comment]
I'm leaning towards the blank one, too.

**Comment by caycefischer** [comment]
Cool, that's what I'll merge then. I'll say it is growing on me :shipit:

**Comment by ghost** [comment]
I prefer the blank one as well.

**Comment by caycefischer** [comment]
Some good stuff coming down the pipe, friends—new header and the beginning of a new “herobox” system for user avatars. Keep your eyes peeled on beta the next couple days, you really gotta play with it.

![newheader min](https://f.cloud.github.com/assets/546495/274201/67c19238-9051-11e2-9301-a8754117e643.gif)

![Screen Shot 2013-03-19 at 12 30 35 AM](https://f.cloud.github.com/assets/546495/274195/2fa3dcee-9051-11e2-9c94-ca6dc4cd336d.png)

![Screen Shot 2013-03-19 at 12 31 37 AM](https://f.cloud.github.com/assets/546495/274196/327cd1e6-9051-11e2-9b82-22874f063f00.png)

![Screen Shot 2013-03-19 at 12 33 22 AM](https://f.cloud.github.com/assets/546495/274198/348811e4-9051-11e2-8f6c-74d86bad8e23.png)

**Comment by switz** [comment]
Nice I dig it! Great job.  

I would maybe combine the + - box into a consistent width, that way every habit aligns properly.

**Comment by wc8** [comment]
I'd suggest diagonal buttons + over -, but that might be too hard to click.

I miss the depth on the power meters, but it does match the rest. XP tooltip is great.

Considering a comment made on difficulty telling the difference between blue and grey shading of the original design: I wonder if the blue/green colors on the settings menu are too similar for people with vision/color problems.

**Comment by caycefischer** [comment]
That's a good point, I'll run it through some tools and ask some
sight-impaired friends. I'm not 100% happy with the blue/grey/teal yet.

**Comment by caycefischer** [comment]
As much as my design ocd would love symmetrical habits regardless of
whether or not it's a plus or a plus and minus...... I don't think we can
do that without sacrificing usability. We need those big guy targets and
clear buttons. Any more ideas friends?

**Comment by wildcate** [comment]
I like the shaded boxes for clicking - I'm just a fan of the classical checkbox design. But the blank one works for me too - I :+1: that the hashed box looked like pre-completed.

And just a wild idea: Maybe the plus-minus-boxes could have the plus on the left and the minus on the right?

**Comment by wc8** [comment]
Since you mentioned it, the blue/grey/teal of the settings drop-down and the reward boxes look a little dingy next to the other vibrant colors. 

I wonder, further, if the reward boxes could be a color not used by the task system (perhaps a shade of purple—_luxury?_?); something that would contrast well with the coin and fit in with the other colors.

**Comment by Shaners** [comment]
Just wanted to say you're going a great job with all of this stuff Zak!

**Comment by lefnire** [comment]
A mother-effing masterpiece. I flip back and forth with the old design from time to time, and get embarrassed :) When I see people using production, I tell them to use beta instead - "oh wow! This is much prettier!". Just remember that feedback is forward-thinking in the Githubosphere, and what you _aren't_ hearing is the present-thinking - which is :+1: :+1: :+1: .

**Comment by wc8** [comment]
The fact that the forward-thinking suggestions are just fine-tuning (nitpicking) is a backwards complement.

**Comment by caycefischer** [comment]
![lemondance](https://f.cloud.github.com/assets/546495/276879/2ddaf052-90b8-11e2-8080-fb15bc58a82c.gif)

**Comment by wc8** [comment]
Hm... blue from the add new in the level label & username drop down? (Just a thought.)
(Could/should those two items be merged?)
![alpha4](https://f.cloud.github.com/assets/3326182/277158/a9aeaf06-90be-11e2-846b-fc5cd34ed35e.png)

**Comment by caycefischer** [comment]
That's a good idea, combining the menu and user area! :+1:

**Comment by wc8** [comment]
By the way, I like having the username and level next to each other.
Here's another _brainstorming idea_: have the username & level # golden to match the experience bar to increase the visual association of level up & experience points.

**Comment by horusofoz** [comment]
Fix implemented "under task button opacity fixed in ba16338" now causes long text tasks to appear Behind the solid colour task add button rather than seeing through it. Need to shorten text field rather than overlay the task add button. (Excuse ignorance of coding lingo/methods).

**Comment by ghost** [comment]
Perhaps an idea for a more minimal/alternative mobile view, have tiles for every habit/daily/todo and just have it show a big button with just that icon, as I'd mostly use the mobile app to update my progress rather than anything else, and this would make that easier.

**Comment by caycefischer** [comment]
@wc8 fixed the line height issue! PRing to `develop` @lefnire (oops wait no—there will be merge confs. Will do w/ you later)

![Screen Shot 2013-03-20 at 10 31 03 PM](https://f.cloud.github.com/assets/546495/284091/6e313f22-91cf-11e2-8ab5-532e1752a48e.png)

**Comment by wc8** [comment]
@xeross that could be considered after [Task/Reward Icons are implemented](https://trello.com/c/FCVdjdUd)
P.S. I like the idea, maybe it could be considered for a future theme. I'll try to add it to Trello tomorrow. For now, I just made a note on the icon suggestion card.

**Comment by lefnire** [comment]
pushed to prod. our branches were getting to out of sync, and cherry-picking was becoming an issue. @zakkain, could you do your input-field magic on the "restore" tool too?

**Comment by caycefischer** [comment]
Yessir​!

**Comment by caycefischer** [comment]
​I gotta pull from your develop branch and sort out any conflicts, I'll
figure it out.

In the future though, put a class of `option-group` on elements wrapping
inputs, and a class of `option-content` on the input itself.

When I get our proper styles sorted (post-bootstrap hacks) this will become
the default input style across all inputs, but right now I need classes to
win specificity.

**Comment by Shmuel510** [comment]
I just got directed here, after the shock of the new design hit...

Probably my feelings on this are all out of proportion to their actual weight, but the best bit of the original design was being able to <I>put a tick in a box.</I> That is one of life's simple joys, one even a Vogon can appreciate. There is an empty box. You put a tick in it. There is now a tick in the box, and that much is well in the world. <i>Shaded</I> boxes being <i>replaced</I> by ticks isn't nearly as satisfying. (Implied boxes, which seems to be where you're headed above, do seem better than shaded boxes, but they still fall short of the tick-in-box experience.)

But that's just me.

**Comment by caycefischer** [comment]
We're on it, dude! Check the bottom of this thread here, we're going with
the blank box option instead. There's been lots of feedback on the
tick-boxing experience, and an update should appear soon. THanks for the
feedback :)

_– Zachary Kain_

Designer, Developer
Writer, Educator
416-712-8895
zakkain@gmail.com
➝ http://draftingcode.com

**Comment by hayleykilroy** [comment]
I like some of changes, but it kind of looks like a rainbow threw up all over my page.  And everything is HUGE now; before, it was more compact & cleaner.  My eyes simply don't know where to look.  This is maybe partly due to the newness of the design, but I feel like before my eyes were drawn to the text (what I needed to do), and now they're drawn to these huge, brightly colored square buttons that detract from the actual task at hand.  And I have to scroll a lot more to see my whole to-do list, which doesn't help me stay on task.

Maybe just mellow the colors out a bit, and make the buttons smaller?  Looks like some of your earlier versions of the design were more along those lines.  (Sorry to be a negative Nancy--please keep up the good work!)

**Comment by mutedgrey** [comment]
I thought I might provide some feedback as well. It's exciting to see changes being made and just a glance at all the stuff to come listed on trello is awesome.  Thank you so much for the time you've put into improving the site!

Things I like about the new design: 1) Breaking up personal vs. store rewards is great. It's not something I thought of but I really like seeing them separated a little. 2) The way the reward price has a colored background in the reward list looks good and makes it easier to read. 3) It collapses nicely when I shrink my browser window and it's fairly easy to scroll through. 4) Overall, if I lean back and try to take in the overall design, it does look very clean, if also very colorful.

Things I don't like about the new design: 1) The colors feel overwhelming and inconsistent between todos, habits, and dailies. It's confusing for completed habits to turn green but completed dailies turn grey (which I prefer). 2) The green plus and red negative signs were easier to see at a glance regardless of state of completion before rather than the whole box turning that color. I also liked them being round before and I like the circles in your earlier designs. 3) Boxes with text that goes onto a second or third line become ginormous! 4) I preferred the rounded corners a lot more, including in your earlier designs, but I realize that's a personal preference. Maybe it could become a toggled option or setting in the future to choose how they look? Like you don't have enough to do already, haha.

**Comment by ghost** [comment]
![](http://puu.sh/2lw6p.png)

The header gradient seems so out of place. Sexy straight lines do not mix with gradients. I think you should have waited before the whole transition was "done" before putting it on production. Try to push that header update ASAP :shipit:

**Comment by lefnire** [comment]
header, avatars, & checkboxes pushed to prod. @aLaserShark & @Shmuel510 check that out.

**Comment by Shmuel510** [comment]
That works much better for me, and I recant the parenthetical bit of my original complaint. Thanks for the speedy update! :-)

**Comment by ghost** [comment]
Many thanks :) Much smoother.

I edited out all the unneeded email crap...Why was that put there again? Stupid outlook...

**Comment by lefnire** [comment]
Might be helpful to start using polls to handle UI disagreements. I've been doing that in Trello when arguments crop up on features developed. Had to do it on the [Priority Multiplier](https://trello.com/card/priority-multiplier/50e5d3684fe3a7266b0036d6/17) for example because people were up-in-arms about 3, 5, 10, or custom # of options. Vast majority said 3 - settled.

Now, downside is only the people with strong opinions as to what's wrong with the UI will show up here, so it might not be as valuable as it could - but it's worth a shot. 

I'll start it off: 
- [Centered, or Left-justified header](http://www.easypolls.net/poll.html?p=514b791fe4b0ecbcda3728b6)
- [Should we make things smaller?](http://www.easypolls.net/poll.html?p=514b7a47e4b0ecbcda3728b8)
- [Rounded, or Sharpe corners](http://www.easypolls.net/poll.html?p=514b7bfae4b0ecbcda3728bb)
- [Too colorful?](http://www.easypolls.net/poll.html?p=514b7daee4b0ecbcda3728bd)

**Comment by Pandoro** [comment]
Tyler you just said what I was going to mention about the new header ;)

**Comment by sgtorange** [comment]
I agree that the task boxes are too big, especially with the text wrapping for longer named task. Like mutedgrey, I also prefer rounded corner to squared boxes. Simply because the habits,daily,todos, and rewards frames are also square cornered which makes the whole page blend together if you look at it from the top making it harder for the user to quickly navigate between tasks. I believe that habitrpg should be concise and easy to use allowing for more time spent doing actual tasks and less time logging progress.

As for trying to define a style, I suggest coming up with custom border graphics for the category boxes and a more distinctive font set. Kind of out there idea but full undergraphics for the boxes would be sweet, imagine instead of just a light blue box background the rewards section can be sitting on top of elongated full color treasure chest(in 8-bit style of course).

I've been following habitrpg's progress closely for a month and I can't tell you how much I appreciate all the hard work everyone's put in to make this a reality. _round of applaud_ By far the most useful thing i've come across in a while, keep up the good work!

thanks for the polls, very useful XD

**Comment by StanLindsey** [comment]
On the subject of too colourful I think its too bright really, too much white creating a high contrast between the less bright colours.

**Comment by Calae** [comment]
Love the new header! Love that I can see my character, pet, level, and username much better than on the other one. And I like the colors, too. Makes it feel more fun, in my opinion.

**Comment by wc8** [comment]
Would making things smaller be a question of less padding?
@lefnire, what if we put an active polls card in the top of the General Features column on Trello? I'll add it, since it can always be archived.

**Comment by Shmuel510** [comment]
Re: polls... discussion of these issues is currently scattered across platforms (multiple threads on Trello, and here, and possibly other places I haven't found yet?), which I'm only just becoming aware of. I am very impressed that you're managing to stay on top of them all! But I'm not so sure I'll manage to do the same.

Perhaps there's something to be said for having a central polling page with a single link somewhere on the HabitRPG page itself? If you wanted, that central page could have links back to the relevant discussions from which the polls sprang for those who want to delve further or provide more nuanced input.

I realize this would probably be a pain in the neck to maintain, so it's probably an impractical suggestion. On the other hand, you'd be more likely to get a wider range of input? (I would also stress "a single link" above. You don't want to clutter up the page.)

Edited to add: ...and @wc8's comment just beat me to the punch. That sounds like a more easily attainable solution.

**Comment by wc8** [comment]
[Vote for Your Preferences: Links to User Polls](https://trello.com/c/y9gkGHWo), list of questions up for vote, with links to original discussion(s) as well as the poll.

By the way, I like the cheerful shades, though with the combo of size+bright it might be a tad too intense. 
Did we ever hear back from vision impaired point of view?
I just hope we don't end up greyscale and/or black and white. (I ended up quitting an app once partly because the icons were all BW/G and my vision's poor.)

**Comment by lefnire** [comment]
Yeah, let's do the active polls card

**Comment by SyntaxPickax** [comment]
I'm still warming up to the new design.  The colors still seem a bit garish.  I agree that things should be smaller.
Multi-line user names actually obscure the avatar:
![Screen Shot 2013-03-21 at 3 03 39 PM](https://f.cloud.github.com/assets/1245093/288389/08496014-9273-11e2-8797-b0bf6b1188b3.png)

**Comment by wc8** [comment]
@lefnire done: [Active Polls](https://trello.com/c/y9gkGHWo) (same as in my last comment).
I added the polls in. We can remove the listing of previous polls. Alternatively, each poll could be made into a checklist item. Wording, of course can be changed as needed.

**Comment by JellyBean22** [comment]
Since the update my username above my character is just "Anonymous", other than that I love the update!

**Comment by wc8** [comment]
I'm not sure if the multi-line name issue would be solved by a size down. As it is, the font for that isn't particularly big. Maybe the box needs to be wider.

**Comment by ghost** [comment]
I'm stuck with just "Anonymous" above my toon too.

Thanks,

Michael Garcia, a.k.a. The Crazy GM http://www.TheCrazyGM.com

**Comment by antistic** [comment]
I agree. I'd prefer things to be smaller. Regarding colours, I think that they're okay, but they're confusing and I find that the information they convey (basically how much HP I might lose etc.) isn't important enough to warrant such bright colours. I think that the checkbox colour is a little too dark compared to the background of the task, and I'm feeling my eyes being drawn to the checkboxes rather than the tasks.
I preferred the colouring of the plus and minus signs for habits, because I have to stop and think and I'd prefer simple red and green for those.
I agree that it's very white - perhaps maybe make the background of each area (habits, todos etc.) a light grey or something.

That being said, it isn't bad and I quite like it.

Edit: A minor thing, but it would be nice if the dropdown menu from the username was the same width as it.

**Comment by lefnire** [comment]
Anonymous is my own bug, I'll take care of that. Also, I'll fix up charts (they were removed temporarily)

**Comment by nafoster** [comment]
I agree with ApplePi's comment of it being very white. Maybe a light cream color could work as well behind each area.

Nonetheless, I also like it a lot. Very good improvement!

**Comment by lefnire** [comment]
@zakkain so far it's looking like Center, Smaller, Sharp Corners, and 50/50 on Colors

**Comment by Pandoro** [comment]
I'd go for smaller first, then see what will happen to the colors. I really
got used to them, but the fact that they are BIG colors, just makes it a
little worse.

**Comment by wc8** [comment]
Yes. I'm with @Pandoro make it smaller, then see what the feeling is about the colors. 
P.S. then maybe redo the poll.

**Comment by lefnire** [comment]
Also, [round of applause](http://www.easypolls.net/poll.html?p=514b9e4fe4b0ecbcda372909) for @zakkain!

**Comment by ghost** [comment]
Problem cannot see the settings butotn completely and cannot see exp or health
![Screenshot_2013-03-21-19-25-18](https://f.cloud.github.com/assets/3267747/288834/1d471c2c-9287-11e2-9cb0-688f24955113.png)

**Comment by wc8** [comment]
:clap:

**Comment by fsiah2** [comment]
When I change shirts I seem to loose half of my helmet.
![HabitRPG](https://f.cloud.github.com/assets/3936882/289044/399b160c-9292-11e2-95d5-4d71fa622a83.PNG)

**Comment by lefnire** [comment]
That's by design, 2 separate female Armor sets. We should just call it set
1 and 2,instead of showing pink and yellow shirts

**Comment by caycefischer** [comment]
## OK, lots of stuff going on here in this thread.

I'm glad to see the polls underway, lots of great feedback coming in. It seems the general opinion is everything is a little too big—that's something I do a lot and I think I'm going to crank down the padding and whatnot a bit across the board. Tiny-text lovers, though, you're out of luck—we're not going back to tinytown. We will likely have user settings in the near future to control overall UI size, as an accessibility feature.

Unless the poll _really_ reveals a dislike of the left-aligned header, I'm inclined to stick with the left-align as well—it is a lot more robust for the party system, and responsive design. And personally, I think the page composition as a whole is better for it.

Since the round/sharp corners poll is in a dead heat, it's **officially a non-issue** and we'll roll with whatever serves the overall interface and aesthetic we have going forward. Probably square, folks.

### Your servant has spoken.

Expect fast and furious hotfixes & tweaks over the weekend. The interface still very much in active development and everything may be subject to change.

![imightbeofservice](https://f.cloud.github.com/assets/546495/289067/d90a963a-9293-11e2-9a42-af7e1552710e.gif)

**Comment by nafoster** [comment]
@lefnire I agree with the label idea, but as long as it doesn't clutter the UI, why not have the label and the icon displaying the shirts? Having the UI stay visual but informative would be best.

**Comment by horusofoz** [comment]
Another reason for the left aligned header is the proposed side bar for NPC/Action launchers

**Comment by caycefischer** [comment]
​That's a new one to me, but sounds rad. So yeah, lots of compelling
reasons for a left-align. Hard to unseat that decision.

**Comment by lefnire** [comment]
Deleted my comment, the show must go on. You have my sword

**Comment by Veovis-Muaddib** [comment]
Once we get smaller boxes I think I'll like this new style a lot.  Especially since the header is no longer static, as I use the site from both a tablet and a desktop browser.

Some notes and requests:
- On my tablet the site is rather unpleasant in portrait orientation.  The health and experience bars are approximately two or three pixels wide, and the new scrollable boxes are not usable until long after the browser's final loading spinner has vanished, and are difficult to interact with once they do work.  [Nexus 7, Chrome 25, Android 4.2.2]
- On mobile devices, a reload of the page is required on changing orientation, as the current mode (boxes or columns) sticks around until a reload.  Especially bad when switching from landscape (Usable columns) to portrait. (Good luck using those columns)  Though I've been somewhat taking advantage of that by loading in portrait and switching to landscape, as landscape boxes on that size screen are rather nice to use. [Nexus 7, Chrome 25, Android 4.2.2]
- I believe phones and tablets (At least when in portrait) would be much easier to use with a tab bar, something akin to what iOS uses at the bottom of many applications.  (Google image search gets me this as I don't have an iOS device handy at the moment: http://bit.ly/ZQ1D1q )  This might be something that is easier to handle in the mobile application, but I thought it was worth mentioning here.
- UI Size usability options would be wonderful.  If I can control size I like to have things much smaller than they would be if I designed them for others.  And those with vision impairments would of course appreciate larger size options.
- Since a size toggle is on the table, what do you think of a dark theme option?  I have a lot of trouble with sites like HabitRPG (And Github for that matter) because of how bright they are.  Not an issue using them for a few minutes at a time, but something like HabitRPG that I use as a todo list, I want visible all the time.

**Comment by caycefischer** [comment]
@lefnire I really don't want to impose my opinion over the will of the crowd. But it really boils down to personal preference or minor grievance for a lot of people, whereas there are some solid technical and design reasons for one direction over the other. So my reasoning is that I'd be inclined to go with the (perhaps unpopular) choice that best benefits the project. _shrug_

@Veovis-Muaddib (baller name by the way) a dark theme would be significantly down the road, as we're first going to work on the "Retro"/game-inspired skin that full-on goes retro-pixel art mode and takes the UI into Legend of Zelda territory. Maybe other skins in the future. But here's a freebie: if  you're having trouble with predominantly bright websites, definitely check out [f.lux](stereopsis.com/flux/), it's a life-saver for me.

**Comment by wc8** [comment]
If the alignment of the header part of a work in progress (sidebar) it might be too soon to get user feedback on it. (Besides that the poll's only been up for some hours.) 

A couple of thoughts on some details that might affect the centered/left question:
- whether or not you have party members will affect the balance of the header; not having them may contribute to the left alignment seeming odd. 
- the user menu on the right and the username-level label on the left suggest a symmetry that the meters don't follow, if that's consolidated, maybe it would help with the overall balance. Or, perhaps if the meters weren't as long they might seem less like they belong in the middle. On the other hand, maybe it's just something we need time to get used to. 

_If_ the user menu (settings, etc.) gets combined with the username & level label, perhaps the undo button could move to the other corner?

**Comment by Veovis-Muaddib** [comment]
@zakkain Thanks!  Two of my favorite fictional characters.

An actual RPG skin is definitely a better place to sink time into for this project specifically, and I look forward to seeing what comes out of it.  But the moment there's a dark theme- that's where I'm headed.

Still, as long as I keep it hidden away most of the time this theme works for me.  Keep up the good work, a request for a dark theme is just that, and I in no way want to devalue what you're doing at the moment.

I already use F.lux on all desktop and iOS 4.0+ devices, it really does help.  Though still waiting on a port/looking for an Android version I like, and can't use it on the old iPod Touch.

**Comment by horusofoz** [comment]
Strangely enough I found if I scrolled down my task list I can't see the
hover over avatar status info for myself or party members. Header is
static/sticky but need to be at top of list for the statuses to show.

**Comment by wildcate** [comment]
Regarding @Shmuel510's comment about the eye being drawn towards the colour spot, not the task: I hear you. Maybe we could try having the line around the box a little thicker after all, and in the same colour as the bright spot? Also, yes yes yes to the deep satisfaction of putting a tick into a box that begged to be ticked. Just feels soooo goooood.

And regarding the polls in general: how about putting "new polls - vote on HabitRPG's design here!" as a link in a box similar to the "new stuff" thing? I only heard about the polls because I check github, I keep logged in into Habit so I never see the splash screen. And then give some time to users to actually do that poll ;)

Left vs. centered header: That's one thing where I think the sudden change caused much of the polling against it. Header alignment, in my experience, takes some time to get used to or someone who is trained to step back, mentally, and check both alignments side by side sort of impassionately. I think that folks will adjust. Or, if folks prefer it centered after all, maybe you can just shift the left-aligned header a bit to the right? So it's still leftish, but indented?

 Finally, the "too colourful" poll is not on the trello card.

Anyway, thanks for all the hard work - and for the effort of listening to the gazillion bits of feedback!

**Comment by minirich** [comment]
When i saw the new design today, i was shocked, i thought the CCS in not loading correct.
I am a Software developer, and i hate eye candy, but this design is clunky (that was the nicest word that came to my mind). If my 4 year old daughter draws a lego building, even there the blocks look smoother.
I tried it with the latests Chrome and Firefox, and i must say, i am not pleased, and i think it was a step in the wrong direction.

**Comment by hajnalmadar** [comment]
This new design may look good on big monitors, I guess the colorful boxes balance the big white space of the background, but on my 12" monitor it looks very cluttered, distractingly colorful, the boxes are too big because they don't fit the small space and most of them occupy two lines. I also preferred the coloring of the plus and minus signs for habits, and the centered alignment of the header. 
If I had the option to go back to the old design, I would do it without hesitation - it wasn't as pretty as this new one, but it wasn't distracting, it served its function very well. That is why I agree with minirich about the wrong direction, functionality is the most important (for me, at least).

**Comment by SebCorbin** [comment]
@minirich :+1:

**Comment by Alys** [comment]
@minirich and @hajnalmadar express my opinions exactly. It really does look eye-wrenching on a small screen. (Still a great tool though. :))

**Comment by skully** [comment]
![8952046d-f386-452d-8f2d-1c766380a305](https://f.cloud.github.com/assets/175585/290649/21a63c6c-92eb-11e2-8db5-d43ebd788cfa.jpg)
I ⌛ unicode. How can I help on this problem?
I would like to get back to old design. Can it be an option?

**Comment by horusofoz** [comment]
- For users on small screens - Have you tried zooming out? I do so on my smallish work PC screen and at 75% it's beautiful.
- For users on tablets, etc, please keep in mind we plan to introduce a mobile app so this interface is not the primary one planned for you mobile device.

**Comment by wc8** [comment]
For those worried about the brightness and/or size, let's wait and see what comes of any [size reduction](https://github.com/lefnire/habitrpg/issues/585#issuecomment-15277146), and see where the colors are then.

@zakkain for the people wanting more of a box, an idea: 
![image](https://f.cloud.github.com/assets/3326182/291378/d9fb2544-9303-11e2-951b-bd2917b9d6de.png)
(Poor version, size wise and shade wise, but just to get the idea across.)

Another idea, possibly a bad one, but here goes: 
- could the New [Habit/Daily/Todo/reward] box take the place of the column header? 
- alternatively, perhaps the header text could be made a smaller, to give us more room for items. 
  ![header idea](https://f.cloud.github.com/assets/3326182/291468/f52482f0-9305-11e2-8de3-5b6281feacfd.png)

**Comment by horusofoz** [comment]
"_could the New [Habit/Daily/Todo/reward] box take the place of the column
header?_"

+1 If headers were a shade or two lighter inside the text field

**Comment by Veovis-Muaddib** [comment]
@horusofoz Yeah, I'm aware, but as I have no idea of the timeframe on the mobile application I figure I might as well keep reporting issues I encounter.  :)

@wc8 I'm not sold on merging the headers and the new item boxes, especially if they retain the word new.  I'd be more apt to support it if they still said "Habits", "Dailies", etc... but I'm still not sure.

However, I replied because I do like the idea of moving the gold count.  Not sure where I'd move it exactly, but it could be nice giving it a new home.

**Comment by mutedgrey** [comment]
@wc8 I'm not really a fan of the shaded perimeter in the check off spot. It just feels like trying to fix something only moderately problematic (desiring an easier to see check mark or ticking off boxes) by making it a lot worse (it doesn't really address what people wanted and it looks worse).

I do like the moved coins, though I don't think it's a big deal if they stay put either. Boxes with two lines of text definitely look better than they did before. On a sidenote, I voted on the polls but I didn't see one about color listed. Am I missing something?

**Comment by wc8** [comment]
My suggestions here are just brainstorming, not a firm proposal.
I think [too colorful?](http://www.easypolls.net/poll.html?p=514b7daee4b0ecbcda3728bd) was added later. Glad you pointed it out. I'll make sure it's on the list of polls.

**Comment by wc8** [comment]
Oops: corrected link: [too colorful?](http://www.easypolls.net/poll.html?p=514b7daee4b0ecbcda3728bd)

@Veovis-Muaddib What if, the new boxes left off the word "New"? Perhaps not clear enough. 
(BTW, I don't like the font, font thickness or size I used, but it was just a mockup anyway.)

**Comment by Veovis-Muaddib** [comment]
@wc8 The concern of course, is that with "New Habit" it's not entirely clear that the column below is filled with habits as well.  With "Habits" it's more clear but I'm still not sold.

Yeah, I get that it's a mockup, you should see my mockups if you think THAT needs a disclaimer.  :P

=-=-=-=-=-=-=-=-=-=

As far as the left aligned vs centered health and exp goes, I'd say add a divider line (like at the top of the four columns) immediately to the right of the bars.  With the bars simply occupying the space left of center without a divider line they draw attention to the imbalance, but with a divider afterward there is no imbalance.

That also has the side effect of causing users to visually group their avatar and their bars together.  Then if they are in a party the party members are clearly separate from the user's own health/exp.

**Comment by lefnire** [comment]
@zakkain I added a "Hotfixes" Trello card for organization: https://trello.com/card/re-design-hotfixes/50e5d3684fe3a7266b0036d6/258

And added a "Design" column for any other design tasks

**Comment by StanLindsey** [comment]
Can I suggest one thing for when tasks turn multi-line;

![Quicksuggestion](https://f.cloud.github.com/assets/3317882/294525/089f582a-93f4-11e2-8d2f-37a94bb30baf.jpg)

**Comment by caycefischer** [comment]
You know Stan, I thought that would look really shitty so I never even tried it.... But it looks GREAT actually, much better than how it is now. Consider it done. Good call man.

**Comment by Pandoro** [comment]
I would be a little cautious here, I think it will indeed look shitty if we
go to three lines. Or four etc. Then  I guess what we have right now looks
better :x ?

**Comment by wc8** [comment]
Maybe a mock-up would help see if it would look ugly or not.
I do like the 2 line version.
Also, how would it look/work if it was changed for 2 lines and not for 4 lines?

**Comment by StanLindsey** [comment]
@pandoro - maybe we need a line limit? There are notes on each task for a reason.

**Comment by StanLindsey** [comment]
With 4 lines of text; I think it still looks better personally. 

![habitsuggestion2](https://f.cloud.github.com/assets/3317882/294618/4ff2d380-9400-11e2-8be9-98fdcb2e9b22.jpg)

**Comment by StanLindsey** [comment]
And I thought i'd add it in context:

![suggestion3](https://f.cloud.github.com/assets/3317882/294623/d257e108-9400-11e2-9496-948e869dd7c4.jpg)

**Comment by Pandoro** [comment]
Personally, I like the square best, I agree that the little lighter space
that is under the square if you only have two lines looks a bit funny, but
I also think it looks a bit funny if the square is no longer a square.

**Comment by Shmuel510** [comment]
I'm on Team Square. Ideally with @wc8's border suggestion from yesterday.

**Comment by lefnire** [comment]
Poll?

**Comment by wc8** [comment]
Can we do images in a poll?

**Comment by Shmuel510** [comment]
A poll sounds good. But one refinement first... one thing I suspect @SlappyBag is reacting to is the text flowing underneath the square, which I would agree doesn't look so great. How about this? Square retained, borders omitted, space under the square left open:
![square and whitespace](https://f.cloud.github.com/assets/3933734/294800/aabf09ca-9417-11e2-9a4b-236fd76c1554.png)
(Top is present behavior, bottom is proposal)

**Comment by Veovis-Muaddib** [comment]
@SlappyBag's bar all the way down looks incredibly nice, but the checkmark is the issue.  Centered vertically might end up rather ugly for three or more lines, top-aligned would look unbalanced.  But a checkable box as discussed earlier in the thread might look nice at the top of the full-height bar.

@Shmuel510's suggested text flow change would work well in the absence of full-height boxes.

I think full-height bar looks better for small amounts of text, two or three lines.  Square box with better text flow looks better with larger amounts of text, four or more lines.  Comparing them overall I think full-height bar always looks nice, while box only looks nice for larger amounts of text.

HabitRPG's general use case, from what I've seen, is to have several smaller titled items, some with an added description as appropriate.  I think full-height bar fills that use case better, but I feel it would rely on a dedicated checkbox.

**Comment by caycefischer** [comment]
I'll have some iterations on the options talked about here up this
afternoon, and we'll see how it looks. For my money, I like your solution
muaddib: a box at the top of a full height colour bar.

On Sunday, March 24, 2013, Veovis-Muaddib wrote:

> @SlappyBag https://github.com/SlappyBag's bar all the way down looks
> incredibly nice, but the checkmark is the issue. Centered vertically might
> end up rather ugly for three or more lines, top-aligned would look
> unbalanced. But a checkable box as discussed earlier in the thread might
> look nice at the top of the full-height bar.
> 
> @Shmuel510 https://github.com/Shmuel510's suggested text flow change
> would work well in the absence of full-height boxes.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/lefnire/habitrpg/issues/585#issuecomment-15353185
> .

## 

_– Zachary Kain_

Designer, Developer
Writer, Educator
416-712-8895
zakkain@gmail.com
➝ http://draftingcode.com

**Comment by StanLindsey** [comment]
Solution: Have a char limit for the task title. You dont need to write an essay for the title of the task do you? Thats what the notes input is for.

**Comment by wildcate** [comment]
Please don't put in a char limit. That would be a total killer for me. Even
if it's ugly (or at least not as sleek as a one-line only thing) I
sometimes have tasks that have a long name, and I don't want to split it up
and put half of it into the notes. A char limit that would limit tasks to
one line is one that would make the list totally useless for me.

Yes, I like words and I might tend to longer to-do list names. I also
prefer having stuff in the list itself instead of having to mouse-over the
notes icon to see what it actually is. And finally I'm also German and that
means I can have a task that is called "Nähmaschinenreparaturservice
anrufen" and it would totally suck if a char limit would kill that...

**Comment by Alys** [comment]
@SlappyBag , I don't think a char limit is a good solution. It takes what is currently a once-off developers' problem (how to make the application look good with long items) and turns it into an ongoing users' problem (how to work around a length limitation by shrinking their task titles or moving text into a note when they'd really rather have that text always visible). To a user who is not aware of this design issue, a char limit will seem like an arbitrary and irritating anti-feature.

I sometimes use URLs as my task titles (with nothing in the note) - the task is simply to read a web page. A char limit would kill that. I could put the URL in the note and then have a few words as the task title, but that would make entering the task significantly more time-consuming, and would also make clicking on the link less convenient.

The best applications are generally those that have excellent usability. Feature limitations for the purposes of ease of development tend to work against that.

**Comment by caycefischer** [comment]
​I agree, a character limit that's anything under tweet-length would be
really restrictive, and even then I'm not sure there's a point to it.

**Comment by mutedgrey** [comment]
I was just wondering, would there be a way to let more users know the polls exist and they could weigh in on these choices? They're kind of a bit difficult to find unless you're actively looking into bugs, changes, or wish to complain about an issue and there's very few votes registered. The poll's responses might be different and more accurate if it were accessed by the wider userbase. Even if it weren't linked on the main page or in update news (though that'd probably be open to the most people), temporarily linking it in the FAQ or something might be helpful?

**Comment by mgopher** [comment]
Agree with @mutedgrey re:polls.
Any solution without the text wrapping will get my vote, it's making me want to abbreviate all my tasks so I don't have to look at it :P I think the bar all the way down looks niiiiiice.

**Comment by StanLindsey** [comment]
When I meantioned char limit i meant for about 3/4 lines. But if people would prefer without then no problem for me =D

**Comment by caycefischer** [comment]
Hey all,

I'm working on the checkbox thing, but such a simple visual change is rarely easy where CSS is concerned. This will get rolled into some more general bugfixes and tweaks in a short while, it requires some changes to the markup and pretty major changes to the styling. Stay tuned, folks.

**Comment by caycefischer** [comment]
so far we got this:

![Screen Shot 2013-03-25 at 9 49 08 AM](https://f.cloud.github.com/assets/546495/298111/d81f97ea-9552-11e2-82fc-5891b9382ff6.png)

(the last one is in mid-hover even though you can't see the cursor)

**Comment by Shmuel510** [comment]
I love the proposed return of an actual box to tick... but I once again protest having the box disappear after it's ticked. Why not change the second of those to match the third?

**Comment by Shaners** [comment]
The HP and EXP bars are sexy by the way, sometimes I just mouse over them for fun now.

**Comment by caycefischer** [comment]
@SHmuel510 We ca​n maybe get that back when we switch up the way the CSS
works (Unicode characters and pseudo-elements), right now having it overlay
like that is janky on a lot of browsers so it's just a appear/replace.

**Comment by Shmuel510** [comment]
@zakkain Pardon the stupid question, but why overlay instead of using a unified tick-in-box graphic or character? ☑

**Comment by caycefischer** [comment]
I have ​yet to find a font that

a) has a glyph for that character that I think is remotely decent
b) is common to android/iphone/mac/windows/linux so that the glyph actually
shows up instead of a blank box.

:(

**Comment by mutedgrey** [comment]
@zakkain I don't think what you have so far would satisfy the ticking boxes enjoyment. I don't know how to describe this and maybe it's just me. The box and check mark are so faint and lackluster, the whole thing feels kind of weak. Before, if memory serves, the tick box had some shading to give it some depth and ticking it off felt like you were actively filling the tick mark. I'm sorry I couldn't offer more help from the technical side of things!

**Comment by Shmuel510** [comment]
I've been looking around at web fonts... currently we're using the version of Lato from Google Web Fonts, which does indeed leave most of the glyphs out. However, Adobe has its own free web font service, at http://html.adobe.com/edge/webfonts/ , which offers the full version of Lato and others, including the glyphs for both unticked and ticked boxes. (You can enter "U+2610 (☐), U+2611 (☑)" as sample text to try 'em out.) Might that help?

**Comment by caycefischer** [comment]
Shmuel510, that's AMAZING. I had no idea the adobe edge version had more
glyphs. Looking into this asap!

**Comment by Shmuel510** [comment]
Pleased to be of service, and thank you! :-)

(Note the bit about "Subsets" in the Edge documentation; I'm guessing we'd need to specify "all" for this.)

**Comment by caycefischer** [comment]
I'm noticing that the glyph actually isn't any different for many of the
fonts, it's falling back to the basic Lucida Unicode or whatever it is.

I'm gonna roll with a custom icon font as planned, and that'll get us
well-made glyphs for these characters we want (checks, boxes, all the other
icons on the site). I'm looking at what's available on
http://icomoon.io/app/ and http://fontello.com/ right now.

**Comment by Shmuel510** [comment]
Yeah. On looking more closely, it's the same in _all_ of them; only the line height changes. Hmmph.

(And I hadn't known about those two sites. Coolness.)

**Comment by mgopher** [comment]
Not sure if you guys changed something, but since yesterday I no longer have working check boxes in Chrome (just [?]'s).  In FF they look fine (although seem to be a little below centre, unless I'm imagining it) so I guess something is not as browser compatible as you thought?

**Comment by mgopher** [comment]
Re: design choices, not very fussed about a persistent check box, but the coloured bar extending all the way down looks awesome.  Now I look at it, could we have the same for habits?  not sure how that would work with habits that have both + & - though.  I would be happy with one under the other, but you guys might disagree.  Alternatively perhaps make them a little smaller, so the darker coloured bar is about as wide as the blue 'rewards' values.   Having the solid bar down the left side of every item would make the whole page look way more unified.

I'm excited to log in and see the improvements every day! amazing stuff guys.

**Comment by caycefischer** [comment]
Habits are tough, I've been thinking about that. It would be _extremely_
difficult (as far as I can think of how to do it) to have the + and –
switch​ from side-by-side to vertical when the habit goes onto more than
one line.

Anyone have any cool ideas for this? It's pretty good right now to be
honest but a more elegant solution could be out there somewhere.

**Comment by mgopher** [comment]
I was thinking maybe something like this horrible, 30 second mock-up pleasedon'tjudgemeI'msupposedtobewritingmythesis:
![shittyhabitmockup](https://f.cloud.github.com/assets/3960155/301525/43dc926e-95d8-11e2-8802-59ec717c3130.png)
although probably with the +/- centred if there is only one.
Even if the +/- is the same size it is now, the darker section might as well go all the way down the left side like with To Do's.

**Comment by Veovis-Muaddib** [comment]
![mockup](https://f.cloud.github.com/assets/2359968/302479/ee5c0f0c-9606-11e2-92ef-17eb9d993e25.png)

Obviously the text doesn't fit an up/down system.

For longer than one line, have a divider at the bottom of the down button.

Centering, symbols, and line thickness/darkness are all up for grabs, this was a minute in Paint.NET.

**Comment by StanLindsey** [comment]
I think i'd prefer the tickbox/tick to be centered vertically if the coloured background is going to resize.

Also this is how it shows up my end atm
![cropped](https://f.cloud.github.com/assets/3317882/302587/24d25e3a-960a-11e2-9658-b2f9f369ddad.jpg)

**Comment by Alys** [comment]
I like Veovis-Muaddib's diagonal +/- a LOT! It looks very classy.

**Comment by StanLindsey** [comment]
The +/- might not work so well with touch, accidently hitting minus would be annoying. Plus it stands out as the only diagonal design element we currently would have.

**Comment by Veovis-Muaddib** [comment]
@SlappyBag:

Centered horizontally and top aligned such that it is centered on a one line task is how I think it'd look nicest myself, but that's just personal preference.  Mobile Chrome is showing it exactly that way and it looks very nice.  Desktop Chrome is aligning it weirdly.

I figure having it at the top gives a single place across all of the tasks to check.  You don't have to scan for the midpoint and look, it's always in the top left corner.

![Screenshot_2013-03-26-06-20-57](https://f.cloud.github.com/assets/2359968/302719/edff97b4-960f-11e2-8045-98a8acafd65b.png)
(On looking closer it's a bit left of center, but still.)

Yeah, I figured that mockup could be difficult to press on mobile before I even made it, but the request was for more ideas to solve this issue.

This is the only major diagonal, true.  But it's not the only diagonal on the page by far.  I see six in just the image in this post.  :P

**Comment by wc8** [comment]
If it could be worked out for a touchscreen, the +/- looks the nicest so far. 
Touchscreen-wise, perhaps some padding in between would help?

It would be the only diagonal, but it currently is the only column with two boxes, and only some of the items have occupy two box-worths of space.

**Comment by antistic** [comment]
Hi, I'm also getting that question mark in a box. It seems to only happen on chrome, though, not firefox, and also affecting the "re-roll" option too. However I don't particularly like how it appears on firefox. I think it would look better just being some sort of image (and therefore looking the same between browsers), if possible...
I like the diagonal idea too, it's better visually (though I'm concerned about how easy it is to click, like most other people).

**Comment by cgardner** [comment]
I've noticed that the input fields for the lists is partially covered by the add button.  It would be nice if I could see the last few characters of my to-do items.

Changing the width on .addtask-field input to 89% seems to fix the issue.

**Comment by heyimzsa** [comment]
Agreed and I would like to be able to use urls in my info box for my todo list.

**Comment by Shaners** [comment]
Was wondering what the new checklist tasks might look like as they progress so quickly mocked this:
![projectprogress](https://f.cloud.github.com/assets/2374703/621037/b3ee57a4-cef4-11e2-85a1-ef6b081fd05b.gif)
Probably going to try another version with no gradient since it's a little out of place.

**Comment by mutedgrey** [comment]
@Shaners I have wanted something like that for so long! That is so exciting! My only critique is that I don't think a rainbow gradient works either. Maybe it could change color, but as a solid color once you pass a certain percentage. For instance, going on the gradient you have there, when you have 0-24% complete it's solid red, 25-49% complete is solid yellow, 50-74% is green, 75-99% is blue, maybe 100% a special separate color? I look forward to seeing how the checklist tasks work!

**Comment by StanLindsey** [comment]
I think the solid colours would suit Habits current style more. That said I like the ring because it differentiates between standard list items and projects/checklists.
